packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-table.tsx

```diff
@@ -0,0 +1,251 @@
+    if (opts.includes(name)) {
+      urlOptions.push({
+        value: value,
+        name: name as UrlOption['name'],
+      });
+    }
+  });
+  return urlOptions;
+}
+
+function UrlOptionsTable({
+  updateConnectionFormField,
+  connectionStringUrl,
+}: {
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionStringUrl: ConnectionStringUrl;
+}): React.ReactElement {
+  const [options, setOptions] = React.useState<Partial<UrlOption>[]>([]);
+  const [errorMessge, setErrorMessge] = React.useState('');
+
+  useEffect(() => {
+    const options: Partial<UrlOption>[] = getUrlOptions(connectionStringUrl);
+    if (!options.length) {
+      options.push({ name: undefined, value: undefined });
+    }
+    setOptions(options);
+  }, [connectionStringUrl]);
+
+  const addUrlOption = () => {
+    setErrorMessge('');
+    // Use case: User clicks on `Add url option` button and then clicked again without completing existing entry.
+    // Don't add another option in such case
+    if (options.find(({ name }) => !name)) {
+      setErrorMessge('Please complete existing option.');
+      return;
+    }
+    const newOptions = [...options, { name: undefined, value: undefined }];
+    setOptions(newOptions);
+  };
+
+  const updateUrlOption = (
+    currentName?: UrlOption['name'],
+    name?: UrlOption['name'],
+    value?: string
+  ) => {
+    setErrorMessge('');
+    const indexOfUpdatedOption = options.findIndex(
+      (option) => option.name === currentName
+    );
+    const option = {
+      name,
+      value,
+    };
+    const newOptions = [...options];
+    newOptions[indexOfUpdatedOption] = option;
+    setOptions(newOptions);
+
+    if (name) {
+      updateConnectionFormField({
+        type: 'update-search-param',
+        // If the user is selecting key for the first time, currentName is undefined and we choose the selected value.
+        currentKey: currentName ?? name,
+        // If the keys are same, then user is changing value. So we set newKey to undefined.
+        newKey: currentName === name ? undefined : name,
+        value: value ?? '',
+      });
+    }
+  };
+
+  const deleteUrlOption = (name?: UrlOption['name']) => {
+    setErrorMessge('');
+    const indexOfDeletedOption = options.findIndex(
+      (option) => option.name === name
+    );
+    const newOptions = [...options];
+    newOptions.splice(indexOfDeletedOption, 1);
+    setOptions(newOptions);
+    if (name) {
+      updateConnectionFormField({
+        type: 'delete-search-param',
+        key: name,
+      });
+    }
+  };
+
+  return (
+    <>
+      <Table
+        data-testid="url-options-table"
+        data={options}
+        columns={[
+          <TableHeader
+            key={'name'}
+            label="Key"
+            className={tableHeaderStyles}
+          />,
+          <TableHeader
+            key={'value'}
+            label="Value"
+            className={tableHeaderStyles}
+          />,
+        ]}
+      >
```

Comment:
Do we need to have a table here now? It seems that these table headers are duplicating the input placeholders anyway, so I'm not sure if having them adds anything (and it seems to complicate the implementation quite a lot)

Reply:
Will bring this thing up with Claudia and @Anemy.